src/super_gradients/training/models/kd_modules/kd_module.py

```diff
@@ -25,13 +25,13 @@ class implementing Knowledge Distillation logic as an SgModule
 
     """
 
-    def __init__(self, arch_params: HpmStruct, student: SgModule, teacher: torch.nn.Module, run_teacher_on_eval=False):
+    def __init__(self, arch_params: HpmStruct, student: SgModule, teacher: torch.nn.Module, run_teacher_on_eval=False, ignore_teacher_input_adapter: bool = False):
```

Comment:
This is a very sloppy undocumented fix.

Lets just separate the input adaptor from the teacher module, and edit KDModule's forward.

Reply:
Changed in 4e118831a32538f03e3ede7f566a90ba7c5d8d7a